utils/client.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+
+"""WCB CLI client.
+"""
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import urllib2
+import urlparse
+
+ENTITIES = {
+    '/': ['', ('',), (), ()],
+    'achievement': ['achievement', ('', 'id'), ('name', 'distance'), ('id',)],
+    'achievements': ['achievements', ('team', 'achievement'), ('team_id', 'achievement_id'),
+                     ('id',)],
+    'cause': ['causes', ('', 'id'), ('name',), ('id',)],
+    'donor': ['donor', (), (), ()],
+    'donors': ['donors', (), (), ()],
+    'event': ['events', ('', 'id'), ('name', 'description', 'start_date', 'end_date', 'team_limit',
+                                     'team_building_start', 'team_building_end'), ('id',)],
+    'locality': ['localities', ('', 'id'), ('name',), ('id',)],
+    'participant': ['participants', ('fbid',), ('fbid',), ('fbid',)],
+    'record': ['records', ('id',), ('date', 'distance', 'participant_id', 'source_id'), ()],
+    'source': ['sources', ('', 'id'), ('name',), ('id',)],
+    'sponsor': ['sponsor', (), (), ()],
+    'sponsors': ['sponsors', (), (), ()],
+    'team': ['teams', ('', 'id'), ('name',), ('id',)],
+}
+
+CRUD = {
+    'c': 'POST',
+    'r': 'GET',
+    'u': 'PATCH',
+    'd': 'DELETE',
+}
+
+
+def construct_url(endpoint):
+    """Build a URL from the scheme, server, and port parameters."""
+    scheme = os.environ.get('SCHEME', 'https')
+    domain = os.environ.get('SERVER', 'akf-causes.subshell.org')
+    port = os.environ.get('PORT', '')
+    if port:
+        netloc = '{}:{}'.format(domain, port)
+    else:
+        netloc = domain
+    url = urlparse.urlunparse((scheme, netloc, endpoint, '', '', ''))
+    return url
+
+
+def get(entity, endpoint, action, url, args):
+    """Construct a GET request for the given entity and parameters."""
+    if not ENTITIES[entity][1]:
+        print('Method Not Allowed')
+        return
+    if entity == 'achievements':
+        if not args:
+            print('Query achievements with "team_id=<team_id>" \
+                   or "achievement_id=<achievement_id>"')
+            return
+        if args[0].startswith('team'):
+            url += '/team/{}'.format(args[0].split('=')[1])
+        elif args[0].startswith('achievement'):
+            url += '/achievement/{}'.format(args[0].split('=')[1])
+        else:
+            print('Query achievements with "team_id=<team_id>" \
+                   or "achievement_id=<achievement_id>"')
+            return
+        return urllib2.Request(url)
```

Comment:
The `return f()` is not particularly pythonic for the `void` return type.  I do get that its less verbose.